bagelbids/tests/test_cli_pheno.py

```diff
@@ -182,3 +194,44 @@ def test_diagnosis_and_control_status_handled(runner, test_data, tmp_path):
     assert "diagnosis" not in pheno["hasSamples"][1].keys()
     assert "diagnosis" not in pheno["hasSamples"][2].keys()
     assert pheno["hasSamples"][2]["isSubjectGroup"] == "purl:NCIT_C94342"
+
+
+def test_get_assessment_tool_availability(test_data):
+    """
+    Ensure that subjects who have one or more missing values in columns mapped to an assessment
+    tool are correctly identified as not having this assessment tool
+    """
+    with open(test_data / "example6.json", "r") as f:
+        data_dict = json.load(f)
+    pheno = pd.read_csv(test_data / "example6.tsv", sep="\t")
+    test_columns = ["tool_item1", "tool_item2"]
+
+    assert are_not_missing(test_columns, pheno.iloc[0], data_dict) is False
+    assert are_not_missing(test_columns, pheno.iloc[2], data_dict) is False
+    assert are_not_missing(test_columns, pheno.iloc[4], data_dict) is True
+
+
+def test_assessment_data_are_parsed_correctly(runner, test_data, tmp_path):
+    runner.invoke(
+        bagel,
+        [
+            "--pheno",
+            test_data / "example6.tsv",
+            "--dictionary",
+            test_data / "example6.json",
+            "--output",
+            tmp_path,
+            "--name",
+            "my_dataset_name",
+        ],
+    )
+
+    with open(tmp_path / "pheno.jsonld", "r") as f:
+        pheno = json.load(f)
+
+    assert pheno["hasSamples"][0].get("assessment") is None
```

Comment:
Is there a straightforward way we could use subject labels for these assertions rather than indices? Right now figuring out which subject in the example corresponds to which index in `"hasSamples"` requires a bit of mental work including accounting for number of sessions. Based on my playing around with a list/array of dicts locally I feel like indexing using subject labels might introduce more python logic than is worthwhile to add here, but let me know if you have any thoughts or ideas

Reply:
no, that's exactly the call I made while writing the test as well. That said: I agree, I want this test to be more natural language like. there is probably a way to do this without adding much complexity if we think about it a little bit